engine/lib/phx/src/engine/engine.rs

```diff
@@ -483,4 +521,276 @@ impl Engine {
         Metric::reset();
         Profiler::end();
     }
+
+    // === Render Thread Control ===
+
+    /// Start the multithreaded render system.
+    ///
+    /// This transfers the GL context to a dedicated render thread.
+    /// All GL operations will then go through command mode.
+    /// Returns true if successfully started.
+    #[bind(name = "StartRenderThread")]
+    pub fn start_render_thread_ffi(&mut self) -> bool {
```

Comment:
I think we don't need this methods at all here. Just add a method that returns RenderContext to Lua and expose its methods via FFI. Same as we do with HmGui, EventBus and TaskQueue.